lib/vanagon/component.rb

```diff
@@ -140,10 +141,13 @@ def get_build_dir
     #
     # @param workdir [String] working directory to put the source into
     def get_sources(workdir)
-      @sources.each do |source|
-        cur_source = Vanagon::Component::Source.source(source.url, { :ref => source.ref, :sum => source.sum }, workdir)
-        cur_source.fetch
-        cur_source.verify
+      sources.each do |source|
```

Comment:
https://github.com/mckern/vanagon/blob/97bd1c4a8b697d9f3f0fd1f236ab9fae5589d96f/lib/vanagon/component.rb#L14

`attr_accessor` is the magic you're looking for.